python/Path.py

```diff
@@ -430,6 +438,19 @@ def x(y):
 
         return False
 
+    def nextWaypointReached(self, positionLatlon):
+        previousWaypointLatlon = self._latlons[self._nextWaypointIndex - 1]
+        nextWaypointLatlon = self._latlons[self._nextWaypointIndex]
+
+        return self.waypointReached(positionLatlon, previousWaypointLatlon, nextWaypointLatlon)
+
+    # assumes there are at least 2 elements in self._latlons
+    def lastWaypointReached(self, positionLatlon):
+        previousWaypointLatlon = self._latlons[len(self._latlons) - 2]
+        nextWaypointLatlon = self._latlons[len(self._latlons) - 1]
+
+        return self.waypointReached(positionLatlon, previousWaypointLatlon, nextWaypointLatlon)
```

Comment:
Issues with this implementation:
* Returns true every time the simulation starts
* Assumes that there are at least 2 elements in `self._latlons`